common/server_protocolhandler.cpp

```diff
@@ -765,3 +751,24 @@ void Server_ProtocolHandler::resetIdleTimer()
     lastActionReceived = timeRunning;
     idleClientWarningSent = false;
 }
+
+bool Server_ProtocolHandler::isUserPermittedToJoinRoom(const QString roomPermission, const QString roomPrivilegeLevel)
```

Comment:
Can you make this method take `userInfo` and mark it static so it's more easily tested with unit tests?